### PR TITLE
Add support for deeplink to open files directly

### PR DESCRIPTION
**Pull request by VishalNehra**
#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [ ] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [ ] `./gradlew assembledebug`
- [ ] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by TranceLove** [comment]
I want to make sure I'm doing the test correctly.

I made a guest only share on my Linux Mint desktop after reading [this](https://www.techrepublic.com/article/how-to-create-a-passwordless-guest-share-in-samba/), then I made a simple HTML with hyperlink saying
```html
<!DOCTYPE html>
<html>
<head>
</head>
<body>
    <p><a href="http://teamamaze.xyz/open_file?path=smb%3A%2F%2F192.168.191.191%2Fpublic%2Fimage.jpg">Open Image</a></p>
</body>
</html>
```
i.e. true path is smb://192.168.191.191/image.jpg

On Pixel 2 emulator running Android 11, HTML was loaded locally using HTML viewer and Chrome. Both cases could trigger Amaze successfully, but then I saw exception thrown
```java
2022-06-08 22:58:36.734 944-8248/? W/System.err: java.lang.NumberFormatException: For input string: "168.191.191"
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at java.lang.Integer.parseInt(Integer.java:615)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at java.lang.Integer.parseInt(Integer.java:650)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.google.android.textclassifier.AnnotatorModel.nativeAnnotate(Native Method)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.google.android.textclassifier.AnnotatorModel.annotate(AnnotatorModel.java:179)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.android.textclassifier.TextClassifierImpl.generateLinks(TextClassifierImpl.java:310)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.android.textclassifier.DefaultTextClassifierService.lambda$onGenerateLinks$2(DefaultTextClassifierService.java:97)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.android.textclassifier.DefaultTextClassifierService.lambda$onGenerateLinks$2$DefaultTextClassifierService(Unknown Source:0)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.android.textclassifier.-$$Lambda$DefaultTextClassifierService$yXvzteEXitEHOLvc8n1LDN1uTOg.call(Unknown Source:4)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
2022-06-08 22:58:36.734 944-8248/? W/System.err:     at java.lang.Thread.run(Thread.java:923)
```

**Comment by VishalNehra** [comment]
I see the smb path is url encoded, try with following link 
http://teamamaze.xyz/open_file?path=smb://192.168.191.191/public/image.jpg

**Comment by TranceLove** [comment]
Tested Cloud plugin with Google Drive, it's OK. Remember though the actual path needs to be `gdrive:///` prefix. Didn't test OneDrive and Dropbox though, but I think Cloudrail plugin would complain the same.

With d2c3574 there is a possible pitfall, which is also visible as I test with Pixel 2 emulator running Android 11.

I have a SMB connection and SSH connection, with SSH connection's Uri.getAuthority() would return the same value as input SMB URI embedded into the deeplink. i.e.

`ssh://user@127.0.0.1` vs `smb://user:password@127.0.0.1` while deeplink is `http://teamamaze.xyz/open_file?path=smb://user@127.0.0.1/file.jpg`

If the SSH connection is created before the SMB connection, running `SshClientUtils.formatPlainServerPathToAuthorised()` would drop to the SSH connection first.

At the second case after removing the SSH connection, if the SMB connection in Amaze is created with password, SshClientUtils.formatPlainServerPathToAuthorised() may end up comparing authorities like `user:password@127.0.0.1` vs `user@127.0.0.1` which will imply none of the existing connections would fit the input deeplink URI.

**Comment by VishalNehra** [comment]
@TranceLove Agreed.. that's a pitfall but I guess we're okay for the time being. 
Did you happen to test anonymous connection, that is, without username / password?

**Comment by TranceLove** [comment]
> @TranceLove Agreed.. that's a pitfall but I guess we're okay for the time being. Did you happen to test anonymous connection, that is, without username / password?

If talking about anonymous connection, no regression found so far. I even thought for SMB anonymous connections, and SSH connections with key authentication, we can live without `SshClientUtils.formatPlainServerPathToAuthorised()`, for that SSH connections can already automatically look for public key fingerprint from the database, then look for private key for authentication.

File: README.md [hunk 1]
```diff
@@ -49,6 +49,14 @@ Support
...
 <a href="https://liberapay.com/Team-Amaze/donate"><img src="https://upload.wikimedia.org/wikipedia/commons/2/27/Liberapay_logo_v2_white-on-yellow.svg" alt="Liberapay" width="80px" ></a>  
 Or buy the [Cloud Plugin](https://play.google.com/store/apps/details?id=com.filemanager.amazecloud) supports Google Drive™, Dropbox, OneDrive and Box accounts.  
+Try our app - [Amaze File Utilities](https://play.google.com/store/apps/details?id=com.amaze.fileutilities)
+1. List videos / images / music documents in your device in a interactive UI where you're able to group / sort and quickly jump to any headers.
+2. Open videos / images / music / documents (pdf / docx / epub) with inbuilt player.
+3. Share / delete / cast on your tv
+4. Analyse internal storage for junk files, duplicate files, large videos / old downloads / screenshots or recordings.
+5. Analyse and group images between memes, low light / blurry / selfies / group pics.
+7. Transfer files directly between two mobile devices on same wifi network using high speed peer to peer network
+8. Gesture support in image / video player, play in background, picture in picture mode, download subtitles within the player.
```
Comment (VishnuSanal) [inline README.md:59]: Do you mind taking the trouble to move this to a new PR? :)

**Comment by VishnuSanal** [comment]
I see some workflow files getting changed & reverted back. Why?

File: README.md (hunk 1 above)
Comment (VishalNehra) [reply README.md:59]: Let it be.

**Comment by VishalNehra** [comment]
> I see some workflow files getting changed & reverted back. Why?

I wanted to generate apk file for test purpose. No real reason.

**Pull request merged**